agenta-web/src/components/TestSetTable/TestSetTable.tsx

```diff
@@ -163,14 +166,21 @@ const TestSetTable: React.FC<TestSetTableProps> = ({mode}) => {
         try {
             let response
             if (mode === "create") {
-                response = await createNewTestSet(appName, testSetName, rowData)
+                if (!testSetName) {
+                    setIsModalOpen(true)
+                } else {
+                    response = await createNewTestSet(appName, testSetName, rowData)
+                    if (response.status === 200) {
+                        router.push(`/apps/${appName}/testsets`)
+                    }
+                }
             } else if (mode === "edit") {
                 response = await updateTestSet(testset_id, testSetName, rowData)
```

Comment:
You need the same modal here! I tried updating a test set and removing the name and I was able to do so. It should not be the case

Reply:
Okay, okay!